Model/articles.model.js

```diff
@@ -36,15 +36,47 @@ exports.updateArticleById = (article_id, inc_votes) => {
     })
 }
 
-exports.selectArticles = () => {
-    
-    return db.query(`SELECT articles.author, articles.title, articles.article_id,articles.topic, articles.created_at, articles.votes, CAST(COUNT(comments.article_id) AS int) as comment_count 
+exports.selectArticles = (sort_by = 'created_at', order = 'desc', topic) => {
+    const validSortBy = ['created_at', 'author', 'title', 'article_id', 'topic', 'votes', 'comment_count']
+
+    let qryStr = `SELECT articles.author, articles.title, articles.article_id,articles.topic, articles.created_at, articles.votes, CAST(COUNT(comments.article_id) AS int) as comment_count 
     FROM articles 
     LEFT JOIN comments 
-        ON articles.article_id = comments.article_id
+        ON articles.article_id = comments.article_id 
     GROUP BY articles.article_id
-    ORDER BY created_at DESC
-    `).then(results => {
+    `
+
+    if(sort_by){
+        if(validSortBy.includes(sort_by)) {
+         qryStr += ` ORDER BY ${sort_by}`
+        } else {
+            return Promise.reject({status: 404, msg: 'property doesnt exist'})
+        }
+    }   
+
+    if(!(order === 'asc' || order === 'desc')) {
+        return Promise.reject({status: 400, msg: 'bad request'})
+    }
+    
+
+    if(order === 'desc'){
+        qryStr += ` DESC`
+    }
+    
+    
+    return db.query(qryStr).then(results => {
+        
+        if(topic){
+            const topicCheck = results.rows.filter(article => {
+            return article.topic === topic
+            })
+            if(topicCheck.length === 0){
+                return Promise.reject({status: 404, msg: 'property doesnt exist'})
+            } else {
+                return topicCheck  
+            }
+        }
```

Comment:
We should not be filtering in JavaScript. We can filter topics with a WHERE clause in SQL, and an array with the topic in it. This will bring back anything that has the topic in it, and if nothing comes back, the topic does not exist